src/app/shared/modules/context-menu/context-menu.module.ts

```diff
@@ -0,0 +1,21 @@
+import { CommonModule } from "@angular/common";
+import { NgModule } from "@angular/core";
+import { FormsModule } from "@angular/forms";
+import { IonicModule } from "@ionic/angular";
+
+import { ContextMenuComponent } from "./context-menu.component";
+import { ContextMenuService } from "./context-menu.service";
+
+@NgModule({
+  imports: [CommonModule, FormsModule, IonicModule],
+  declarations: [ContextMenuComponent],
+})
+/**
+ * The contextMenuModule handles tracking events such as right-click and text selection that can
+ * be used to trigger a custom context menu, allowing for registered actions to be made available.
+ * This should be imported into main app.module to allow tracking across the app
+ */
+export class ContextMenuModule {
```

Comment:
Most of the logic for handling context menus is in the `contextMenuService`, however we are combining this into a core `module` as it also includes a default display component (used to render the menu buttons). This is the first time we have specified a core `module` as opposed to `service`, however something I think likely we will continue to do in the future where relevant (previously things have been separated to have shared components and shared services that refer to each other but not collated into a module)